kaldb/src/main/java/com/slack/kaldb/clusterManager/ReplicaCreatorService.java

```diff
@@ -39,23 +48,46 @@ protected void startUp() throws Exception {
     metadataStoreService.awaitRunning(DEFAULT_START_STOP_DURATION);
 
     replicaMetadataStore = new ReplicaMetadataStore(metadataStoreService.getMetadataStore(), true);
-    replicaMetadataStore.addListener(replicaNodeListener());
+
+    snapshotMetadataStore =
+        new SnapshotMetadataStore(metadataStoreService.getMetadataStore(), true);
+    snapshotMetadataStore.addListener(this::createReplicasForUnassignedSnapshots);
+
+    // we need to fire this at startup, so any snapshots that were created while we were offline are
+    // handled
+    createReplicasForUnassignedSnapshots();
   }
 
   @Override
   protected void shutDown() throws Exception {
     LOG.info("Closing replica create service");
 
     replicaMetadataStore.close();
+    snapshotMetadataStore.close();
 
     LOG.info("Closed replica create service");
   }
 
-  private KaldbMetadataStoreChangeListener replicaNodeListener() {
-    return () -> {
-      List<ReplicaMetadata> replicaMetadataList = replicaMetadataStore.getCached();
-      LOG.debug(
-          "Change on replica metadata, new replica metadata count {}", replicaMetadataList.size());
-    };
+  private void createReplicasForUnassignedSnapshots() {
+    snapshotMetadataStore
```

Comment:
Add logging before and after this process runs. 

Add a counter so we know how many times this process has run. 

Also, it may be better to add a metrics to measure how long a single execution of this process took to run.

Reply:
Done. Switched to a timer so that we get both the count and the time information.